pkg/operator/csidriveroperator/driverstarter.go

```diff
@@ -0,0 +1,177 @@
+package csidriveroperator
+
+import (
+	"context"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorapi "github.com/openshift/api/operator/v1"
+	openshiftv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/cluster-storage-operator/pkg/csoclients"
+	"github.com/openshift/cluster-storage-operator/pkg/generated"
+	"github.com/openshift/cluster-storage-operator/pkg/operator/csidriveroperator/csioperatorclient"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/controller/manager"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/staticresourcecontroller"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/client-go/dynamic"
+)
+
+const (
+	conditionsPrefix = "CSIDriverStarter"
+	infraConfigName  = "cluster"
+)
+
+// This CSIDriverStarterController starts CSI driver controllers based on the underlying cloud.
+// It does not install anything by itself, only monitors Infrastructure
+// instance and starts ControllerManagers for the particular cloud.
+// It produces following Conditions:
+// CSIDriverStarterDegraded - error checking the Infrastructure
```

Comment:
Same condition regardless the operator/driver?

Reply:
Yes, there is only one `driverstarte` in CSO.